bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVMessageParser.java

```diff
@@ -0,0 +1,334 @@
+                String remoteServerVersion = GoogleTVRequest.encodeMessage(remoteServerVersionSb.toString());
+
+                logger.debug("{} - {} \"{}\" \"{}\" {} {} {}", callback.getThingID(), preamble, model, manufacturer,
+                        androidVersion, remoteServer, remoteServerVersion);
+
+                callback.setModel(model);
+                callback.setManufacturer(manufacturer);
+                callback.setAndroidVersion(androidVersion);
+                callback.setRemoteServer(remoteServer);
+                callback.setRemoteServerVersion(remoteServerVersion);
+
+            } else if (msg.startsWith(DELIMITER_12)) {
+                // Second message on connection from GTV
+                // Login successful
+                callback.sendCommand(
+                        new GoogleTVCommand(GoogleTVRequest.encodeMessage(GoogleTVRequest.loginRequest(5))));
+                logger.info("{} - Login Successful", callback.getThingID());
+                callback.setLoggedIn(true);
+            } else if (msg.startsWith(DELIMITER_92)) {
+                // Third message on connection from GTV
+                // Also sent on power state change (to ON only unless keypress triggers)i
+                // 9203 21 08 02 10 02 1a 11 534849454c4420416e64726f6964205456 20 02 2800 30 0f 38 0e 40 00
+                // --------DD----DD----DD-LEN-SHIELD Android TV
+                // 9203 1e 08 9610 10 09 1a 0d 4368726f6d6563617374204844 20 02 2800 30 19 38 0a 40 00
+                // --------DD------DD----DD-LEN-Chromecast HD
+                // 9203 1a 08 f304 10 09 1a 11 534849454c4420416e64726f6964205456 20 01
+                // 9203 1a 08 8205 10 09 1a 11 534849454c4420416e64726f6964205456 20 01
+                // --------DD------DD----DD-LEN-SHIELD Android TV
+                //
+                // VOLUME:
+                // ---------------DD----DD----DD-LEN-BRAVIA 4K GB------------DD---------DD-MAX---VOL---MUTE
+                // 00 --- 9203 1c 08 03 10 06 1a 0c 42524156494120344b204742 20 02 2800 30 64 38 00 40 00
+                // 01 --- 9203 1c 08 03 10 06 1a 0c 42524156494120344b204742 20 02 2800 30 64 38 01 40 00
+                // 100 -- 9203 1c 08 03 10 06 1a 0c 42524156494120344b204742 20 02 2800 30 64 38 64 40 00
+                // MUTE - 9203 1c 08 03 10 06 1a 0c 42524156494120344b204742 20 02 2800 30 64 38 00 40 01
+
+                String st = "";
+                int length = 0;
+
+                StringBuffer preambleSb = new StringBuffer();
+                StringBuffer modelSb = new StringBuffer();
+                String volMax = "";
+                String volCurr = "";
+                String volMute = "";
+                String audioMode = "";
+
+                int i = 0;
+                int current = 0;
+
+                for (; i < 12; i++) {
+                    preambleSb.append(charArray[i]);
+                }
+
+                st = "" + charArray[i] + charArray[i + 1];
```

Comment:
Are these `"" + ` in all this code really necessary?

```suggestion
                st = charArray[i] + charArray[i + 1];
```

Reply:
Unfortunately yes, I had issues without them. Adding "" + forces the String to build properly when the first variable is by default not a String.